hibernate-core/src/test/java/org/hibernate/orm/test/embeddable/discriminatorvalues/EmbeddableInheritanceTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.embeddable.discriminatorvalues;
+
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@DomainModel(
+		annotatedClasses = {
+				Animal.class,
+				Cat.class,
+				Dog.class,
+				Fish.class,
+				Mammal.class,
+				Owner.class
+		}
+)
+@SessionFactory
+public class EmbeddableInheritanceTest {
```

Comment:
Just a minor nitpick, but we already have a lot of tests regarding embeddable inheritance in the `org.hibernate.orm.test.inheritance.embeddable`. Perhaps you could move this class there and rename it to something more meaningful (i.e. `EmbeddableInheritanceDiscriminatorValuesTest` or something). You could also re-use some of the mappings there, or simply make the ones used here inner classes to make the test a bit cleaner, but I'll leave that up to you.

Reply:
Done ... more or less.